src/mail/editor/MailEditor.ts

```diff
@@ -159,59 +177,69 @@ export class MailEditor implements Component<MailEditorAttrs> {
 
 		// call this async because the editor is not initialized before this mail editor dialog is shown
 		this.editor.initialized.promise.then(() => {
-			this.editor.setHTML(model.getBody())
-			// Add mutation observer to remove attachments when corresponding DOM element is removed
-			new MutationObserver(onEditorChanged).observe(this.editor.getDOM(), {
-				attributes: false,
-				childList: true,
-				subtree: true,
-			})
-			// since the editor is the source for the body text, the model won't know if the body has changed unless we tell it
-			this.editor.addChangeListener(() => model.setBody(replaceInlineImagesWithCids(this.editor.getDOM()).innerHTML))
-			this.editor.addEventListener("pasteImage", ({ detail }: ImagePasteEvent) => {
-				const items = Array.from(detail.clipboardData.items)
-				const imageItems = items.filter((item) => /image/.test(item.type))
-				if (!imageItems.length) {
-					return false
-				}
-				const file = imageItems[0]?.getAsFile()
-				if (file == null) {
-					return false
-				}
-				const reader = new FileReader()
-				reader.onload = () => {
-					if (reader.result == null || "string" === typeof reader.result) {
-						return
-					}
-					const newInlineImages = [createDataFile(file.name, file.type, new Uint8Array(reader.result))]
-					model.attachFiles(newInlineImages)
-					this.insertInlineImages(model, newInlineImages)
+			model.getExternalImageRule().then((externalImageRule) => {
```

Comment:
I am sorry I keep adding suggestions but now it hit me that maybe we should be doing this before creating the dialog? We a already pass whether we should block or not, we could also fetch the rule in `createMailEditorDialog()` and pass it in as an attribute, either way we won't show the dialog until we fetch the rule. What do you think? I think that would more match how we do other things

Reply:
Maybe not before creating it, but before showing it. We need the previous mail that we don't have until the sendMailModel is initialized.  We'll just move things to ```createMailEditorDialog``` and I think that it's fine, I can't see significant changes at least

Reply:
Well we have added `getExternalImageRule()` to `SendMailModel` ourselves and maybe it's not even the right place.
Also by the time we get to `createMailEditorDialog` we already will have `previousMail` or am I wrong?